tests/main/hidden-snap-dir/task.yaml

```diff
@@ -27,76 +93,76 @@ execute: |
     not test -d "$HOME"/.snap/data
 
     echo "Take a snapshot"
-    "$NAME".cmd echo "prev_data" > "$HOME"/snap/"$NAME"/current/data
+    "$NAME".cmd echo "prev_data" > "$HOME/snap/$NAME/current/file"
     # get the snapshot number from the 2nd line (the 1st line is the header)
     snapshot=$(snap save "$NAME" | awk 'FNR == 2 {print $1}')
 
     echo "Write data to user data dirs"
-    "$NAME".echo "data" > "$HOME"/snap/"$NAME"/current/data
-    "$NAME".echo "common" > "$HOME"/snap/"$NAME"/common/common
+    data="old_data"
+    "$NAME".echo "$data" > "$HOME/snap/$NAME/current/file"
+    "$NAME".echo "$data" > "$HOME/snap/$NAME/common/file"
 
     echo "Refresh the snap"
     "$TESTSTOOLS"/snaps-state install-local "$NAME"
 
-    echo "Check snap directory was migrated"
-    test -d "$HOME"/.snap/data/"$NAME"
-    test -d "$HOME"/.snap/data/"$NAME"/common
-    test -d "$HOME"/.snap/data/"$NAME"/x2
-    not test -d "$HOME"/snap/"$NAME"
-
-    echo "Check the env vars point to ~/.snap/data"
-    snapEnv=$("$NAME".env)
-    echo "$snapEnv" | MATCH "SNAP_USER_DATA=$HOME/\.snap/data/$NAME/x2"
-    echo "$snapEnv" | MATCH "SNAP_USER_COMMON=$HOME/\.snap/data/$NAME/common"
-
-    # 'current' symlink is created just before the snap runs, so this check
-    # must come after a snap run
-    if [ "$(readlink "$HOME"/.snap/data/"$NAME"/current)" != "x2" ]; then
-      echo "expected 'current' to point to new revision after refresh"
-      exit 1
-    fi
-
-    echo "Check that the written data is in the new dir"
-    MATCH "common" < "$HOME"/.snap/data/"$NAME"/common/common
-    MATCH "data" < "$HOME"/.snap/data/"$NAME"/x2/data
+    # Check env vars, dirs and data after the migration
+    check_env "x2" ""
```

Comment:
One more super nitpick: the last parameter is not very wasy to read; maybe you could remove it and instead have a `--with-exposed-home` flag when you want to do the check?

Then your shell functions would start like this:

```sh
local CHECK_EXPOSED_HOME="no"
if [ "$1" = "--with-exposed-home" ]; then
    CHECK_EXPOSED_HOME="yes"
    shift  # now $1 becomes the parameter after this
fi
local REV="$1"
```

Reply:
Or have two function: `check_dirs` and `check_dirs_and_exposed_home` (the latter could implicitly call the former)?